tests/system/action/poll/test_vote.py

```diff
@@ -577,3 +596,1207 @@ def test_default_vote_weight(self) -> None:
+                "poll_id": 1,
+                "yes": "0.000000",
+                "no": "0.000000",
+                "abstain": "0.000000",
+            },
+        )
+        self.update_model(
+            "user/1",
+            {
+                "is_present_in_meeting_ids": [113],
+                "group_$113_ids": [1],
+                "group_$_ids": ["113"],
+            },
+        )
+
+    def create_poll(self) -> None:
+        # has to be implemented by subclasses
+        raise NotImplementedError()
+
+    def start_poll(self) -> None:
+        self.update_model("poll/1", {"state": Poll.STATE_STARTED})
+
+    def add_candidate(self) -> None:
+        self.create_model("option/3", {"meeting_id": 113, "poll_id": 1})
+        self.update_model("poll/1", {"option_ids": [1, 2, 3]})
+
+
+class VotePollOS3AnalogYNA(VotePollBaseTestClass):
+    def create_poll(self) -> None:
+        self.create_model(
+            "poll/1",
+            dict(
+                content_object_id="assignment/1",
+                title="test_title_04k0y4TwPLpJKaSvIGm1",
+                pollmethod="YNA",
+                type=Poll.TYPE_ANALOG,
+                state=Poll.STATE_CREATED,
+                meeting_id=113,
+                option_ids=[1, 2],
+                entitled_group_ids=[1],
+            ),
+        )
+
+    def test_vote(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {
+                "value": {
+                    "1": {"Y": "1", "N": "2.35", "A": "-1"},
+                    "2": {"Y": "30", "N": "-2", "A": "8.93"},
+                },
+                "user_id": 1,
+                "id": 1,
+            },
+        )
+        self.assert_status_code(response, 400)
+        assert "poll.vote is not allowed for analog voting." in response.data.decode()
+
+
+class VotePollOS3NamedYNA(VotePollBaseTestClass):
+    def create_poll(self) -> None:
+        self.create_model(
+            "poll/1",
+            dict(
+                content_object_id="assignment/1",
+                title="test_title_OkHAIvOSIcpFnCxbaL6v",
+                pollmethod="YNA",
+                type=Poll.TYPE_NAMED,
+                state=Poll.STATE_CREATED,
+                meeting_id=113,
+                option_ids=[1, 2],
+                entitled_group_ids=[1],
+                votescast="0.000000",
+                votesvalid="0.000000",
+                votesinvalid="0.000000",
+            ),
+        )
+
+    def test_vote(self) -> None:
+        self.add_candidate()
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "Y", "2": "N", "3": "A"}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 200)
+        self.assert_model_exists("vote/1")
+        self.assert_model_exists("vote/2")
+        self.assert_model_exists("vote/3")
+        self.assert_model_not_exists("vote/4")
```

Comment:
Maybe do something like ensuring, that the count of vote objects is 3

Reply:
Done.